src/useBluetoothNotifications.tsx

```diff
@@ -0,0 +1,188 @@
+    characteristicUuid,
+    deviceOptions,
+    onNotification,
+    onServerDisconnect,
+    onError,
+    parser,
+  } = useMemo(() => {
+    return { ...defaultHookOptions, ..._options };
+  }, [_options]);
+
+  const bluetooth: Bluetooth = useMemo(() => {
+    if (!navigator.bluetooth) {
+      const error = new Error("This device is not capable of using bluetooth");
+      if (onError) {
+        onError(error);
+      } else {
+        throw error;
+      }
+    }
+    return navigator.bluetooth;
+  }, [onError]);
+
+  const handleNotifications = useCallback(
+    (event: Event) => {
+      const value = (event.target as BluetoothRemoteGATTCharacteristic).value;
+      if (value && parser) {
+        const parsed = parser(value);
+        if (onNotification) {
+          onNotification(parsed, event);
+        }
+      }
+    },
+    [onNotification, parser],
+  );
+
+  const handleError = useCallback(
+    (error: Error) => {
+      setStatus(BluetoothNotificationsStatus.ERROR);
+      if (onError) {
+        onError(error);
+      } else {
+        throw error;
+      }
+    },
+    [onError],
+  );
+
+  const reset = useCallback(() => {
+    setCharacteristic(undefined);
+    setService(undefined);
+
+    if (server) {
+      server.disconnect();
+      setServer(undefined);
+    }
+
+    setDevice(undefined);
+    setStatus(BluetoothNotificationsStatus.STOPPED);
+  }, [server]);
+
+  const startStream = useCallback(async () => {
+    try {
+      setStatus(BluetoothNotificationsStatus.STARTING);
+      let service: BluetoothRemoteGATTService;
+      let characteristic: BluetoothRemoteGATTCharacteristic;
+
+      const device = await bluetooth.requestDevice(deviceOptions);
+      setDevice(device);
+
+      const server = await device.gatt?.connect();
+
+      if (server) {
+        setServer(server);
+        service = await server.getPrimaryService(serviceUuid);
+        characteristic = await service.getCharacteristic(characteristicUuid);
+        setService(service);
+        setCharacteristic(characteristic);
+
+        await characteristic.startNotifications();
+        setStatus(BluetoothNotificationsStatus.STARTED);
+      } else {
+        setStatus(BluetoothNotificationsStatus.ERROR);
+        throw new Error("No GATT server found");
+      }
+    } catch (error) {
+      if (error.code === USER_CANCEL_ERROR_CODE) {
+        setStatus(BluetoothNotificationsStatus.CANCELLED);
+        return;
+      }
+      handleError(error);
+    }
+  }, [bluetooth, characteristicUuid, deviceOptions, handleError, serviceUuid]);
+
+  const stopStream = useCallback(async () => {
+    try {
+      if (characteristic && device && device.gatt) {
+        setStatus(BluetoothNotificationsStatus.STOPPING);
+        await characteristic.stopNotifications();
+        reset();
+      }
+    } catch (error) {
+      handleError(error);
+    }
+  }, [characteristic, device, handleError, reset]);
+
+  const _onServerDisconnect = useCallback(
```

Comment:
nitpicky but this is the method that is prefixed with `_` but it is not the only "private" method in this hook.  based on other conventions used maybe this should be called `handleServerDisconnect`

Reply:
Good call. "Can't see the forest from the trees" situation on this one I think.